### PR TITLE
BaSD-MS-02: Mongoose/Admins. Add controllers, routes and model (Schema) for Admin entity.

### PR DESCRIPTION
**Pull request by clementinob**
### Admin entity:
_Implemented **Express Router**, new **Controllers Methods**, add folder "`model`" with the schema based on Mongoose library, and Postman Collection._

Request Methods added:
**GET** - "/"
**GET** - "/:id"
**GET** - "/byEmail/:email"
**POST** - "/create"
**PUT** - "/update/:id"
**DELETE** - "/delete/:id"

Controllers added:

- `getAllAdmins`,
- `getAdminByID`,
- `getAdminByEmail`,
- `createAdmin`,
- `updateAdmin`,
- `deleteAdmin`

### CODE REVIEW

File: models/Admins.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+
+const AdminsSchema = new Schema({
+  email: { type: String, required: true },
```
Comment (aquilesmariani) [inline models/Admins.js:6]: Wondering if we should add a password prop for it. Can we do it or didn't we add it for any reason?

File: controllers/admins.js [hunk 2]
```diff
@@ -1,100 +1,82 @@
...
-    if (admin.id === parseInt(req.params.id)) {
-      adminPosition = index;
-      return admin;
-    }
-  });
-  if (!foundedAdmin) {
-    res.status(404).send(`There is no admin with this id: ${req.params.id}`);
+const deleteAdmin = (req, res) => {
+  if (!req.params.id) {
+    return res.status(400).json({ msg: 'You should enter an Admin ID to be deleted' });
```
Comment (aquilesmariani) [inline controllers/admins.js:61]: You could replace the `delete` method and use it like this instead adding multiple `else` or `else if`
```suggestion
const deleteAdmin = (req, res) => {
  if (!req.params.id) return res.status(400).json({ msg: 'You should enter an Admin ID to be deleted' });
  return Admins.findByIdAndDelete(req.params.id, (error, result) => {
    if (error) return res.status(400).json(error);
    if (result === null) return res.status(404).json({ msg: 'Admin ID not found' });
    return res.status(200).json({ msg: `Admin with the id: ${req.params.id} was deleted!` });
  });
};
```

File: index.js [hunk 3]
```diff
@@ -1,14 +1,6 @@
 require('dotenv').config();
 const express = require('express');
 const mongoose = require('mongoose');
-const interviews = require('./controllers/interviews.js');
```
Comment (aquilesmariani) [inline index.js:4]: After you merge with `development`, is it ok to remove all those imports?

File: models/Admins.js (hunk 1 above)
Comment (aquilesmariani) [reply models/Admins.js:6]: Could we think of a way to add a validation to accept only email formats?

**Review by aquilesmariani — CHANGES_REQUESTED** [review]

File: models/Admins.js (hunk 1 above)
Comment (clementinob) [reply models/Admins.js:6]: Done! 

- Password prop has been added with a validation of password.length >= 8 characters. 
- Validation of email with the Schema prop "**match**" and a _RegEx_ of email.

File: controllers/admins.js (hunk 2 above)
Comment (clementinob) [reply controllers/admins.js:61]: Done! It's very much readable now. Applied on every method with "if and else" statements. 👍

File: index.js (hunk 3 above)
Comment (clementinob) [reply index.js:4]: My bad, now only removed the `imports `and `app.get` endpoints of entities that has been updated to the `development ` branch

**Review by aquilesmariani — APPROVED** [review]
Nice refactor on the `if` conditions 👏

**Pull request merged**